### PR TITLE
#99: Added replacement of broken urls

### LINKED ISSUE #99: Check if links of dependencies work and don't add them to dependencies.md if not
Some projects specify broken links. In that cases, we should better not provide a link than adding a broken one.
Solution:
Check if the links work and only add them if so

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #99

### CODE REVIEW

File: src/main/java/com/exasol/projectkeeper/validators/dependencies/DependenciesBrokenLinkReplacer.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package com.exasol.projectkeeper.validators.dependencies;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.exasol.projectkeeper.BrokenLinkReplacer;
+
+public class DependenciesBrokenLinkReplacer {
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/projectkeeper/validators/dependencies/DependenciesBrokenLinkReplacer.java:8]: Do we still write Javadocs or rules were changed?

File: src/main/java/com/exasol/projectkeeper/BrokenLinkReplacer.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Some maven projects define invalid / outdated links to their project homepage. Since PK writes theses links to the
+ * dependencies.md file each project again contains these broken links. Since that's not desirable and breaks the CI
+ * link checker, this class replaces the broken links by working ones.
+ * <p>
+ * This class has a built in mapping for some common broken links but also allows users to define new link mappings in
+ * the pom file.
+ * </p>
+ */
+public class BrokenLinkReplacer {
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/projectkeeper/BrokenLinkReplacer.java:16]: What's the difference between BrokenLinkReplacer and DependenciesBrokenLinkReplacer?

File: src/main/java/com/exasol/projectkeeper/validators/dependencies/DependenciesValidator.java [hunk 3]
```diff
@@ -39,7 +42,9 @@ public DependenciesValidator(final MavenFileModelReader fileModelReader,
     @Override
     public Validator validate(final Consumer<ValidationFinding> findingConsumer) {
         final List<ProjectDependency> dependencies = this.projectDependencyReader.readDependencies(this.pomFile);
-        final String expectedDependenciesPage = new DependencyPageRenderer().render(dependencies);
+        final List<ProjectDependency> dependenciesWithFixedLinks = new DependenciesBrokenLinkReplacer(
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/projectkeeper/validators/dependencies/DependenciesValidator.java:45]: It's strange somehow that you're fixing dependencies in the validate() method. I think this should happen somewhere during building the model after you read the dependencies. Any chances to move this logic?

**Review by AnastasiiaSergienko — CHANGES_REQUESTED** [review]

File: src/main/java/com/exasol/projectkeeper/validators/dependencies/DependenciesBrokenLinkReplacer.java (hunk 1 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/projectkeeper/validators/dependencies/DependenciesBrokenLinkReplacer.java:8]: I just missed it.

File: src/main/java/com/exasol/projectkeeper/BrokenLinkReplacer.java (hunk 2 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/projectkeeper/BrokenLinkReplacer.java:16]: BrokenLinkReplacer in multi purpose just replacing links. 

DependenciesBrokenLinkReplacer replaces links in dependencies

**Review by AnastasiiaSergienko — APPROVED** [review]

**Pull request merged**